app/backend/src/tests/test_bg_jobs.py

```diff
@@ -649,6 +651,105 @@ def test_process_send_onboarding_emails(db):
+
+    # case 2: host left ref (surfer needs an email)
+    # host
+    user3, token3 = generate_user(email="user3@couchers.org.invalid", name="User 3")
+    # surfer
+    user4, token4 = generate_user(email="user4@couchers.org.invalid", name="User 4")
+
+    # case 3: surfer left ref (host needs an email)
+    # host
+    user5, token5 = generate_user(email="user5@couchers.org.invalid", name="User 5")
+    # surfer
+    user6, token6 = generate_user(email="user6@couchers.org.invalid", name="User 6")
+
+    # case 4: neither left ref (host & surfer need an email)
+    # host
+    user7, token7 = generate_user(email="user7@couchers.org.invalid", name="User 7")
+    # surfer
+    user8, token8 = generate_user(email="user8@couchers.org.invalid", name="User 8")
+
+    with session_scope() as session:
+        # case 1: bidirectional (no emails)
+        ref1, hr1 = create_host_reference(session, user2.id, user1.id, timedelta(days=12), surfing=True)
+        create_host_reference(session, user1.id, user2.id, timedelta(days=12), host_request_id=hr1)
+
+        # case 2: host left ref (surfer needs an email)
+        ref2, hr2 = create_host_reference(session, user3.id, user4.id, timedelta(days=12), surfing=False)
+
+        # case 3: surfer left ref (host needs an email)
+        ref3, hr3 = create_host_reference(session, user6.id, user5.id, timedelta(days=8), surfing=True)
+
+        # case 4: neither left ref (host & surfer need an email)
+        hr4 = create_host_request(session, user7.id, user8.id, timedelta(days=2))
+
+    expected_emails = [
+        ("user4@couchers.org.invalid", "You have 3 days to write a reference for User 3!"),
+        ("user5@couchers.org.invalid", "You have 7 days to write a reference for User 6!"),
+        ("user7@couchers.org.invalid", "You have 12 days to write a reference for User 8!"),
+        ("user8@couchers.org.invalid", "You have 12 days to write a reference for User 7!"),
+    ]
+
+    process_send_reference_reminders(empty_pb2.Empty())
+
+    with session_scope() as session:
+        assert (
+            session.execute(
+                select(func.count())
+                .select_from(BackgroundJob)
+                .where(BackgroundJob.job_type == BackgroundJobType.send_email)
+            ).scalar_one()
+            == 4
+        )
+
+    process_send_reference_reminders(empty_pb2.Empty())
+
+    with session_scope() as session:
+        assert (
+            session.execute(
+                select(func.count())
+                .select_from(BackgroundJob)
+                .where(BackgroundJob.job_type == BackgroundJobType.send_email)
+            ).scalar_one()
+            == 4
+        )
+
+    for _ in range(4):
+        process_job()
+
+    with session_scope() as session:
+        emails = [
+            (email.recipient, email.subject)
+            for email in session.execute(select(Email).order_by(Email.recipient.asc())).scalars().all()
+        ]
+
+        print(emails)
```

Comment:
Should this be removed?

Reply:
Yes, sorry, leftover. Not so critical though since it's just in testing

Reply:
Not critical, but a leftover print threw me off for a bit once